src/Concordium/Client/GRPC2.hs

```diff
@@ -0,0 +1,3222 @@
+    b
+mkSerialize ek = Proto.make (ProtoFields.value .= S.encode ek)
+
+{- |Like 'mkSerialize' above, but used to set a wrapper type whose @value@ field
+ has type @uint64@. The supplied value must be coercible to a 'Word64'.
+ Coercible here means that the value is a newtype wrapper (possibly repeated)
+ of a Word64.
+-}
+mkWord64 ::
+    ( Proto.Message b
+    , Data.ProtoLens.Field.HasField
+        b
+        "value"
+        Word64
+    , Coercible a Word64
+    ) =>
+    a ->
+    b
+mkWord64 a = Proto.make (ProtoFields.value .= coerce a)
+
+-- |Like 'mkWord64', but for 32-bit integers instead of 64.
+mkWord32 ::
+    ( Proto.Message b
+    , Data.ProtoLens.Field.HasField
+        b
+        "value"
+        Word32
+    , Coercible a Word32
+    ) =>
+    a ->
+    b
+mkWord32 a = Proto.make (ProtoFields.value .= coerce a)
+
+{- |Like 'mkWord32', but the supplied value must be coercible to
+ 'Word16'.
+-}
+mkWord16 ::
+    forall a b.
+    ( Proto.Message b
+    , Data.ProtoLens.Field.HasField
+        b
+        "value"
+        Word32
+    , Coercible a Word16
+    ) =>
+    a ->
+    b
+mkWord16 a = Proto.make (ProtoFields.value .= (fromIntegral (coerce a :: Word16) :: Word32))
+
+{- |Like 'mkWord32', but the supplied value must be coercible to
+ 'Word8'.
+-}
+mkWord8 ::
+    forall a b.
+    ( Proto.Message b
+    , Data.ProtoLens.Field.HasField
+        b
+        "value"
+        Word32
+    , Coercible a Word8
+    ) =>
+    a ->
+    b
+mkWord8 a = Proto.make (ProtoFields.value .= (fromIntegral (coerce a :: Word8) :: Word32))
+
+{- |A helper function that serves as an inverse to `mkSerialize`,
+
+ Converts a protocol buffer wrapper type to a native Haskell type
+ which is a member of `Serialize`.
+
+ More concretely, the wrapper type should be of the form
+
+ > message Wrapper {
+ >    bytes value = 1
+ > }
+
+ where the name @Wrapper@ can be arbitrary, but the @value@ field must exist,
+ and it must have type @bytes@.
+-}
+deMkSerialize ::
+    ( Data.ProtoLens.Field.HasField
+        b
+        "value"
+        BS.ByteString
+    , S.Serialize a
+    ) =>
+    b ->
+    a
+deMkSerialize val = case S.decode $ val ^. ProtoFields.value of
+    Left _ -> error "Noo"
+    Right bs -> bs
+
+-- |Like 'deMkSerialize' above, but for Word64 fields.
+deMkWord64 ::
+    ( Data.ProtoLens.Field.HasField
+        b
+        "value"
+        Word64
+    ) =>
+    b ->
+    Word64
+deMkWord64 val = coerce $ val ^. ProtoFields.value
+
+-- |Like 'deMkWord64' above, but for Word32 fields instead.
+deMkWord32 ::
+    ( Data.ProtoLens.Field.HasField
+        b
+        "value"
+        Word32
+    ) =>
+    b ->
+    Word32
+deMkWord32 val = coerce $ val ^. ProtoFields.value
+
+-- |Like 'deMkWord64' above, but for Word16 fields instead.
+deMkWord16 ::
+    ( Data.ProtoLens.Field.HasField
+        b
+        "value"
+        Word32
+    ) =>
+    b ->
+    Word16
+deMkWord16 val = fromIntegral $ shiftR (shiftL (val ^. ProtoFields.value) 16) 16
```

Comment:
What exactly is happening here? You cannot do this. You have to validate that it fits into Word16.